### PR TITLE
Configurations Sections

### PR DESCRIPTION
**Pull request by RotemAmit**
## Status
Ready/In Progress/In Hold (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
I added an explanation about the different possible sections in the integration configuration.

## Screenshots
Paste here any images that will help the reviewer

### CODE REVIEW

File: docs/integrations/yaml-file.md [hunk 1]
```diff
@@ -94,6 +94,70 @@ Integration parameters may be hidden from the XSOAR UI, using the optional `hidd
...
+In the Connect section should be parameters that the customer needs to configure in order to connect to the product. An example for such parameters are Server URL, API Key, etc.
+In the Collect section should be parameters that the customer needs to configure in order to collect information from the product. An example for such parameters are Fetches events radio button, First fetch timestamp, etc.
+In the advanced section should be parameters that are part of advanced usage of the integration. For example fetch queries, etc.
+The sections will look like this in the UI (advanced parameters are hidden unless opening the Advanced Settings  section):
+![Example of the connect and collect sections](/doc_imgs/integrations/sections_connect_collect.png)
+![Example of the advanced section](/doc_imgs/integrations/sections_advanced.png)
+
+In order to add sections to your integration, follow this guide:
+
+1. Add the `sectionorder` key to the yml's root. This key should contain a list of sections available. Currently we support only "Connect",  "Collect" and "Optimize".
```
Comment (BEAdi) [inline docs/integrations/yaml-file.md:110]:
```suggestion
1. Add the `sectionOrder` key to the yml's root. This key should contain a list of sections available. Currently we support only "Connect",  "Collect" and "Optimize".
```
We use it in content this way

File: docs/integrations/yaml-file.md [hunk 2]
```diff
@@ -94,6 +94,70 @@ Integration parameters may be hidden from the XSOAR UI, using the optional `hidd
...
+2. Add the `section` key to each parameter in the configuration, with one of the sections from (1).
+3. If the parameter is to be shown only as part of the advanced settings in the section, add the advanced key with value set to true.
```
Comment (BEAdi) [inline docs/integrations/yaml-file.md:112]:
```suggestion
3. If the parameter is to be shown only as part of the advanced settings in the section, add the `advanced: true` key and value to it.
```

File: docs/integrations/yaml-file.md [hunk 3]
```diff
@@ -94,6 +94,70 @@ Integration parameters may be hidden from the XSOAR UI, using the optional `hidd
...
+
+```yml
```
Comment (BEAdi) [inline docs/integrations/yaml-file.md:114]:
```suggestion
For example:
```yml
```

File: docs/integrations/yaml-file.md [hunk 4]
```diff
@@ -94,6 +94,70 @@ Integration parameters may be hidden from the XSOAR UI, using the optional `hidd
...
 - To hide the parameter in specific content marketplace versions, provide list of marketplace version names (e.g. `xsoar` (XSOAR 6 and XSOAR 8), `xsoar_on_prem` (XSOAR 6), `xsoar_saas` (XSOAR 8 Cloud or On-prem), `marketplacev2` (XSIAM) or `xpanse` (XPANSE))
 
+### Configurations Sections
+As of XSOAR version 8.1 and XSIAM version 1.3, integration's configuration display is divided to sections to help our users configure the instances.
+The sections re-organize the parameters the users have to fill when configuring the instance.
+
+In the Connect section should be parameters that the customer needs to configure in order to connect to the product. An example for such parameters are Server URL, API Key, etc.
+In the Collect section should be parameters that the customer needs to configure in order to collect information from the product. An example for such parameters are Fetches events radio button, First fetch timestamp, etc.
+In the advanced section should be parameters that are part of advanced usage of the integration. For example fetch queries, etc.
+The sections will look like this in the UI (advanced parameters are hidden unless opening the Advanced Settings  section):
```
Comment (BEAdi) [inline docs/integrations/yaml-file.md:104]:
```suggestion
The sections will look like this in the UI, when advanced parameters are hidden unless opening the `Advanced Settings` section:
```

**Review by BEAdi — CHANGES_REQUESTED** [review]
Great work!
I think we should add all of the connect and collect params from the guideline document, so contributors will most likely find what they are looking for here and won't have to ask.

File: docs/integrations/yaml-file.md [hunk 5]
```diff
@@ -94,6 +94,108 @@ Integration parameters may be hidden from the XSOAR UI, using the optional `hidd
...
+  - Traffic Light Protocol Color
+- Advanced Collect Params:
+  - Events Fetch Interval
+  - Mirroring parameters
+  - Any additional filters / Non mandatory parameters
+  - Indicator Expiration Method
+  - Feed Fetch Interval
+  - Bypass exclusion list
+  - Create relationships
+  - Bypass exclusion list
```
Comment (BEAdi) [inline docs/integrations/yaml-file.md:136]:
```suggestion
  - Indicator expiration method
  - Feed fetch interval
  - Bypass exclusion list
  - Create relationships
  - Any additional filters / Non mandatory parameters
```
`Bypass exclusion list` was written twice

File: docs/integrations/yaml-file.md [hunk 6]
```diff
@@ -94,6 +94,108 @@ Integration parameters may be hidden from the XSOAR UI, using the optional `hidd
...
+  - Username
+  - Password
+  - API Key
+  - Other mandatory parameters
+- Advanced Connect Params:
+  - Any additional filters / Non mandatory parameters
+  - Trust any certificate (not secure)
+  - Use system proxy settings
+  - Log Level
+  - Run on Single engine
```
Comment (BEAdi) [inline docs/integrations/yaml-file.md:116]:
```suggestion
  - Trust any certificate (not secure)
  - Use system proxy settings
  - Log level
  - Run on single engine
  - Any additional filters / Non mandatory parameters
```

**Review by BEAdi — APPROVED** [review]
Great!

File: docs/integrations/yaml-file.md [hunk 7]
```diff
@@ -94,6 +94,107 @@ Integration parameters may be hidden from the XSOAR UI, using the optional `hidd
 - To hide the parameter in all marketplaces (XSOAR, XSOAR_SAAS, XSOAR_ON_PREM, XSIAM), use a boolean `true`.
 - To hide the parameter in specific content marketplace versions, provide list of marketplace version names (e.g. `xsoar` (XSOAR 6 and XSOAR 8), `xsoar_on_prem` (XSOAR 6), `xsoar_saas` (XSOAR 8 Cloud or On-prem), `marketplacev2` (XSIAM) or `xpanse` (XPANSE))
 
+### Configurations Sections
+As of XSOAR version 8.1 and XSIAM version 1.3, integration's configuration display is divided to sections to help our users configure the instances.
```
Comment (ShirleyDenkberg) [inline docs/integrations/yaml-file.md:98]:
```suggestion
As of Cortex XSOAR version 8.1 and Cortex XSIAM version 1.3, an integration's configuration display is divided to sections to help our users configure the instances.
```

File: docs/integrations/yaml-file.md [hunk 8]
```diff
@@ -94,6 +94,107 @@ Integration parameters may be hidden from the XSOAR UI, using the optional `hidd
...
+The sections re-organize the parameters the users have to fill when configuring the instance.
+
+In the advanced section should be parameters that are part of advanced usage of the integration.
```
Comment (ShirleyDenkberg) [inline docs/integrations/yaml-file.md:101]:
```suggestion
The Advanced section should contain parameters that are part of advanced usage of the integration.
```

File: docs/integrations/yaml-file.md [hunk 9]
```diff
@@ -94,6 +94,107 @@ Integration parameters may be hidden from the XSOAR UI, using the optional `hidd
...
+In the Connect section should be parameters that the customer needs to configure in order to connect to the product.
```
Comment (ShirleyDenkberg) [inline docs/integrations/yaml-file.md:102]:
```suggestion
The Connect section should contain parameters that the customer needs to configure in order to connect to the product. 
```

File: docs/integrations/yaml-file.md [hunk 10]
```diff
@@ -94,6 +94,107 @@ Integration parameters may be hidden from the XSOAR UI, using the optional `hidd
 - To hide the parameter in all marketplaces (XSOAR, XSOAR_SAAS, XSOAR_ON_PREM, XSIAM), use a boolean `true`.
 - To hide the parameter in specific content marketplace versions, provide list of marketplace version names (e.g. `xsoar` (XSOAR 6 and XSOAR 8), `xsoar_on_prem` (XSOAR 6), `xsoar_saas` (XSOAR 8 Cloud or On-prem), `marketplacev2` (XSIAM) or `xpanse` (XPANSE))
 
+### Configurations Sections
+As of XSOAR version 8.1 and XSIAM version 1.3, integration's configuration display is divided to sections to help our users configure the instances.
+The sections re-organize the parameters the users have to fill when configuring the instance.
+
+In the advanced section should be parameters that are part of advanced usage of the integration.
+In the Connect section should be parameters that the customer needs to configure in order to connect to the product. 
+- Connect Prams:
```
Comment (ShirleyDenkberg) [inline docs/integrations/yaml-file.md:103]:
```suggestion
- Connect Parameters:
```

File: docs/integrations/yaml-file.md [hunk 11]
```diff
@@ -94,6 +94,107 @@ Integration parameters may be hidden from the XSOAR UI, using the optional `hidd
...
+  - Name
+  - Server URL / URL Address
+  - Classifier / Incident Type / Mapper section
+  - Username
+  - Password
+  - API Key
+  - Other mandatory parameters
+- Advanced Connect Params:
```
Comment (ShirleyDenkberg) [inline docs/integrations/yaml-file.md:111]:
```suggestion
- Advanced Connect Parameters:
```

File: docs/integrations/yaml-file.md [hunk 12]
```diff
@@ -94,6 +94,107 @@ Integration parameters may be hidden from the XSOAR UI, using the optional `hidd
...
+  - Trust any certificate (not secure)
+  - Use system proxy settings
+  - Log level
+  - Run on single engine
+  - Any additional filters / Non mandatory parameters
+
+In the Collect section should be parameters that the customer needs to configure in order to collect information from the product.
```
Comment (ShirleyDenkberg) [inline docs/integrations/yaml-file.md:118]:
```suggestion
The Collect section should contain parameters that the customer needs to configure in order to collect information from the product.
```

File: docs/integrations/yaml-file.md [hunk 13]
```diff
@@ -94,6 +94,107 @@ Integration parameters may be hidden from the XSOAR UI, using the optional `hidd
...
+- Collect Params:
```
Comment (ShirleyDenkberg) [inline docs/integrations/yaml-file.md:119]:
```suggestion
- Collect Parameters:
```

File: docs/integrations/yaml-file.md [hunk 14]
```diff
@@ -94,6 +94,107 @@ Integration parameters may be hidden from the XSOAR UI, using the optional `hidd
...
+  - Fetch events / Do not fetch radio buttons or Fetch/Do not fetch indicators
+  - Incident Mirroring Direction
+  - First fetch timestamp
+  - Number of incidents/events to fetch per fetch
+  - Do not use by default
+  - Indicator Reputation 
+  - Source Reliability 
+  - Traffic Light Protocol Color
+- Advanced Collect Params:
```
Comment (ShirleyDenkberg) [inline docs/integrations/yaml-file.md:128]:
```suggestion
- Advanced Collect Parameters:
```

File: docs/integrations/yaml-file.md [hunk 15]
```diff
@@ -94,6 +94,107 @@ Integration parameters may be hidden from the XSOAR UI, using the optional `hidd
...
+  - Events Fetch Interval
+  - Mirroring parameters
+  - Indicator expiration method
+  - Feed fetch interval
+  - Bypass exclusion list
+  - Create relationships
+  - Any additional filters / Non mandatory parameters
+
+The Optimize Params are params that are less fit to be in connect or collect section, e.g: Advanced Thresholds, Advanced Queries.
```
Comment (ShirleyDenkberg) [inline docs/integrations/yaml-file.md:137]:
```suggestion
The Optimize Parameters are parameters that do not belong in the Connect or Collect sections, e.g., Advanced Thresholds, Advanced Queries.
```

File: docs/integrations/yaml-file.md [hunk 16]
```diff
@@ -94,6 +94,107 @@ Integration parameters may be hidden from the XSOAR UI, using the optional `hidd
...
+
+
+The sections will look like this in the UI, when advanced parameters are hidden unless opening the `Advanced Settings` section:
+![Example of the connect and collect sections](/doc_imgs/integrations/sections_connect_collect.png)
+![Example of the advanced section](/doc_imgs/integrations/sections_advanced.png)
+
+In order to add sections to your integration, follow this guide:
```
Comment (ShirleyDenkberg) [inline docs/integrations/yaml-file.md:144]:
```suggestion
To add sections to your integration:
```

**Comment by ShirleyDenkberg** [comment]
@BEAdi Doc review completed.

**Pull request merged**